sql/hive/v0.12.0/src/main/scala/org/apache/spark/sql/hive/Shim12.scala

```diff
@@ -241,8 +241,14 @@ private[hive] object HiveShim {
       Decimal(hdoi.getPrimitiveJavaObject(data).bigDecimalValue())
     }
   }
+
+  implicit def prepareWritable(shimW: ShimWritable): Writable = {
+    shimW.writable
+  }
 }
 
+case class ShimWritable(writable: Writable)
```

Comment:
Yes, I think implicits in this case just make the code harder to read.
On Feb 1, 2015 11:32 PM, "Liang-Chi Hsieh" <notifications@github.com> wrote:

> In sql/hive/v0.12.0/src/main/scala/org/apache/spark/sql/hive/Shim12.scala
> <https://github.com/apache/spark/pull/4014#discussion_r23910154>:
>
> >  }
> >
> > +case class ShimWritable(writable: Writable)
>
> If we skip ShimWriteable, we then need to remove implicit from
> prepareWriteable and explicitly call it to do the fixing. Is it better?
> If so, I can do it in this way.
>
> It does not break Hive 12 because we just pass the underlying writable
> object without touching it. We only do the fixing on Hive 13.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/spark/pull/4014/files#r23910154>.
>